pkg/operator/controller/ingress/status_test.go

```diff
@@ -821,51 +820,72 @@ func TestComputeLoadBalancerStatus(t *testing.T) {
 	testCases := []struct {
-		description          string
-		deploymentConditions []appsv1.DeploymentCondition
-		expect               operatorv1.OperatorCondition
+		description string
+		conditions  []operatorv1.OperatorCondition
+		expect      []operatorv1.OperatorCondition
 	}{
 		{
-			description: "deployment available",
-			deploymentConditions: []appsv1.DeploymentCondition{
-				{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionTrue},
+			description: "deployment, dns, and lb available",
+			conditions: []operatorv1.OperatorCondition{
+				{Type: string(appsv1.DeploymentAvailable), Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.DNSReadyIngressConditionType, Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.LoadBalancerReadyIngressConditionType, Status: operatorv1.ConditionTrue},
 			},
-			expect: operatorv1.OperatorCondition{Type: operatorv1.OperatorStatusTypeAvailable, Status: operatorv1.ConditionTrue},
+			expect: []operatorv1.OperatorCondition{{Type: operatorv1.OperatorStatusTypeAvailable, Status: operatorv1.ConditionTrue}},
 		},
 		{
-			description: "deployment not available",
-			deploymentConditions: []appsv1.DeploymentCondition{
-				{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionFalse},
+			description: "deployment not available, but dns and lb available",
+			conditions: []operatorv1.OperatorCondition{
+				{Type: string(appsv1.DeploymentAvailable), Status: operatorv1.ConditionFalse},
+				{Type: operatorv1.DNSReadyIngressConditionType, Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.LoadBalancerReadyIngressConditionType, Status: operatorv1.ConditionTrue},
 			},
-			expect: operatorv1.OperatorCondition{Type: operatorv1.OperatorStatusTypeAvailable, Status: operatorv1.ConditionFalse},
+			expect: []operatorv1.OperatorCondition{{Type: operatorv1.OperatorStatusTypeAvailable, Status: operatorv1.ConditionFalse}},
 		},
 		{
-			description: "deployment availability unknown",
-			deploymentConditions: []appsv1.DeploymentCondition{
-				{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionUnknown},
+			description: "dns not available, but deployment and lb available",
+			conditions: []operatorv1.OperatorCondition{
+				{Type: string(appsv1.DeploymentAvailable), Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.LoadBalancerReadyIngressConditionType, Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.DNSManagedIngressConditionType, Status: operatorv1.ConditionTrue},
+				{Type: operatorv1.DNSReadyIngressConditionType, Status: operatorv1.ConditionFalse},
+				{Type: operatorv1.LoadBalancerReadyIngressConditionType, Status: operatorv1.ConditionTrue},
```

Comment:
You have `operatorv1.LoadBalancerReadyIngressConditionType` twice in this test case.